src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -482,7 +472,7 @@ object AppDefinition extends GeneralPurposeCombinators {
 
   private val complyWithResidencyRules: Validator[AppDefinition] =
     isTrue("AppDefinition must contain persistent volumes and define residency") { app =>
-      !(app.residency.isDefined ^ app.persistentVolumes.nonEmpty)
+      !(app.isResident ^ app.persistentVolumes.nonEmpty)
```

Comment:
AFAIU, `isResident = persistentVolumes.nonEmpty`, therefore, the whole expression always evaluates to `true`, doesn't it?